### DISCUSSION TITLE
Is there a way to specify gprbuild options in the .toml file?

### DISCUSSION THREAD

**Discussion by pmunts** [alire-project/alire#1315] [Q&A]
You can append `gprbuild `options to `alr build` with `-- <option>`.  Is there some way to specify them in the `.toml` file?

**Comment by Fabien-Chouteau** [comment]
No we don't have this at the moment. What options do you want to specify? Probably most, if not all, of them can be added to the `.gpr` file.

**Reply by pmunts** [reply]
The most important one to me is `--config=somefile.cgpr.`  Unlike `--target=sometarget`, there is no corresponding statement in the `.gpr` file.

The `for target use` statement in the `.gpr` file is a pain in the neck because it requires a matching `.cgpr` file `/usr/share/gpr`, which requires root access to install.

**Reply by jere-software** [reply]
> No we don't have this at the moment. What options do you want to specify? Probably most, if not all, of them can be added to the `.gpr` file.

I've recently run into this problem as well.  I want to have the GPR file to default to a different external value when called via `alr build` vs when they simply call `gprbuild` (I have some users who prefer not to use alire, so I want to support both in an mixed environment).  

My intended solution was to add an option to my external value list called "alire" so that doing:
`alr build` actually calls `gprbuild -XBUILD=alire`

I was hoping I could add a property in the TOML with the value `-XBUILD=alire` but couldn't find one.  After some google searching I found this question.  Since the question is still open, I am guessing this is still on the table as an option?

**Reply by mosteo** [reply]
For the particular case of setting externals for the build, you can use the `gpr-set-externals` property. Look for it in https://github.com/alire-project/alire/blob/master/doc/catalog-format-spec.md#release-information.

**Reply by jere-software** [reply]
Thanks!
